bin/_docker.sh

```diff
@@ -53,10 +62,33 @@ docker_build() {
       if [ -n "$DOCKER_BUILDKIT_CACHE" ]; then
         cache_params="--cache-from type=local,src=${DOCKER_BUILDKIT_CACHE} --cache-to type=local,dest=${DOCKER_BUILDKIT_CACHE},mode=max"
       fi
-      log_debug "  :; docker buildx $rootdir $cache_params --load -t $repo:$tag -f $file $*"
+
+      output_params="--load"
+      if [ -n "$DOCKER_MULTIARCH" ]; then
+
+        # Pushing multi-arch images to gcr.io with the same tag that already exists is not possible
+        # The issue is on gcr as pushing the same tag in docker hub works fine
+        # Related issues: https://github.com/eclipse/che/issues/16983, https://github.com/open-policy-agent/gatekeeper/issues/665
+        if (docker buildx imagetools inspect "$repo:$tag"); then
+          echo "Build skipped. Image already exists"
+          exit 0
+        fi
+
+        output_params="--platform $SUPPORTED_ARCHS"
+        if [ -n "$DOCKER_PUSH" ]; then
+          output_params+=" --push"
+        else
+          echo "Error: env DOCKER_PUSH=1 is missing"
+          echo "When building the multi-arch images it is required to push the images to the registry"
+          echo "See https://github.com/docker/buildx/issues/59 for more details"
+          exit 1
+        fi
```

Comment:
Is there a specific reason `DOCKER_PUSH` exists if it's always needed when `DOCKER_MULTIARCH` is set?

For example, if we only set `DOCKER_MULTIARCH=1` in `release.yml`, then we would just always append ` --push` to `output_params`.

From what I can tell, we don't check for `DOCKER_PUSH` in any other case, so it seems like it's a variable that has to be set for no specific reason.

Reply:
the reason is to make it to be more explicit, as @alpeb suggested in his [comment](https://github.com/linkerd/linkerd2/pull/4794#discussion_r461736683)

Reply:
Ah okay that makes sense. I'm good with being more explicit 👍